proxy/coco/src/peer.rs

```diff
@@ -479,6 +495,12 @@ impl Future for Subroutines {
             }
         }
 
+        if self.run_state.status != old_status {
```

Comment:
This feels kind of specialised to me. I feel like it makes sense to just always report the current status and any consumer of these messages can filter on the lack of transitions.

Reply:
For this to make sense you'd need to bind the status to the event, a la: `(status, event)`. The semantics of this is clear to only emit on status transition. Arguably better would be: `(old_status, new_status)`.